lib/tasks/add_missing_asset_audit_records.rake

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Rails task to add missing asset audit records from a CSV file
+namespace :asset_audit do
+  # This rake task is used to add missing asset audit records from a CSV file.
+  # If there are any errors in the CSV file, no records will be inserted.
+  # Usage: rails asset_audit:add_missing_records['/path/to/file.csv']
+  # The CSV file should have the following headers:
+  # barcode, message, created_by, created_at
+  # The message column should have one of the following values:
+  # 'Destroying location', 'Destroying labware'
+
+  desc 'Add missing asset audit records'
+  task :add_missing_records, [:file_path] => :environment do |_, args|
+    required_csv_headers = %w[barcode message created_by created_at]
+
+    # Check if the file path is provided and valid
+    file_path = args[:file_path] == 'nil' ? nil : args[:file_path]
+    if file_path.nil? || !File.exist?(file_path)
+      puts 'Please provide a valid file path'
+      next
+    end
+
+    # Read the CSV file and validate the headers
+    begin
+      csv_data = CSV.read(file_path, headers: true)
+    rescue StandardError => e
+      puts "Failed to read CSV file: #{e.message}"
+      next
+    end
+
+    unless csv_data.headers.length == required_csv_headers.length
+      puts 'Failed to read CSV file: Invalid number of header columns'
+      next
+    end
+
+    # Process the CSV data and create asset audit records data array to insert
+    asset_audit_data = []
+    csv_data.each do |row|
```

Comment:
We are **skipping** bad rows in this loop, instead of stopping the processing. Is this intended?

Reply:
Good question! Now that you've asked it, I think it’s better to stop the processing and return in case of any errors, ensuring uniformity across all validations. Therefore,  for the addition of records to occur, we need to ensure that all rows in the CSV file are error-free. I believe this approach is better.

Reply:
Good question! Now that you've asked it, I think it’s better to stop the processing and return in case of any errors, ensuring uniformity across all validations. Therefore, for the addition of records to occur, we need to ensure that all rows in the CSV file are error-free. I believe this approach is better.
 I will make this change. Thank you.